src/foam/blob/RestBlobService.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package foam.blob;
+
+import foam.core.X;
+
+import java.net.URL;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.nio.ByteBuffer;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.InputStreamReader;
+
+public class RestBlobService
+    extends AbstractBlobService
+{
+  public static final int BUFFER_SIZE = 8192;
+
+  protected String address_;
+
+  public RestBlobService(String address) {
+    this.address_ = address + "/httpBlobService";
+  }
+
+  public RestBlobService(foam.core.X x, String address) {
+    setX(x);
+    this.address_ = address + "/httpBlobService";
+  }
+
+  public String getAddress() {
+    return address_;
+  }
+
+  @Override
+  public Blob put_(X x, Blob blob) {
+    if ( blob instanceof IdentifiedBlob ) {
+      return blob;
+    }
+    HttpURLConnection connection = null;
+    OutputStream os = null;
+    InputStream is = null;
+    IdentifiedBlob result = null;
+    try {
+      URL url = new URL(address_);
+      connection = (HttpURLConnection) url.openConnection();
+      
+      //configure HttpURLConnection
+      connection.setConnectTimeout(5 * 1000);
+      connection.setReadTimeout(5 * 1000);
+      connection.setDoOutput(true);
+      connection.setUseCaches(false);
+
+      //set request method
+      connection.setRequestMethod("PUT");
+
+      //configure http header
+      connection.setRequestProperty("Accept", "*/*");
+      connection.setRequestProperty("Connection", "keep-alive");
+      connection.setRequestProperty("Content-Type", "application/octet-stream");
+
+      //output blob into connection
+      long chunk = 0;
+      long size = blob.getSize();
+      long chunks = (long) Math.ceil((double) size / (double) BUFFER_SIZE);
+      Buffer buffer = new Buffer(BUFFER_SIZE, ByteBuffer.allocate(BUFFER_SIZE));
+      os = connection.getOutputStream();
+
+      while ( chunk < chunks ) {
+        buffer = blob.read(buffer, chunkOffset(chunk));
+        byte[] buf = buffer.getData().array();
+        os.write(buf, 0, (int) buffer.getLength());
+        buffer.getData().clear();
+        chunk++;
+      }
+
+      os.flush();
+
+      if ( connection.getResponseCode() == HttpURLConnection.HTTP_OK ) {
+        is = connection.getInputStream();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+        String json = "";
```

Comment:
I would parse this the same way we do in src/foam/nanos/http/ServiceWebAgent.java

CharBuffer buffer = CharBuffer.allocate(65535);
reader.read(buffer);
buffer.rewind();
parserString(buffer.toString());

Later we're going to let the parser read directly from an Reader, but in the meantime we might as well be consistent so anybody grepping the code will find both instances.